equinox/nn/_conv.py

```diff
@@ -134,21 +174,34 @@ def __init__(
         self.out_channels = out_channels
         self.kernel_size = kernel_size
         self.stride = stride
-        if isinstance(padding, int):
-            self.padding = tuple((padding, padding) for _ in range(num_spatial_dims))
-        elif isinstance(padding, Sequence) and len(padding) == num_spatial_dims:
-            if all_sequences(padding):
-                self.padding = tuple(padding)
-            else:
-                self.padding = tuple((p, p) for p in padding)
-        else:
-            raise ValueError(
-                "`padding` must either be an int or tuple of length "
-                f"{num_spatial_dims} containing ints or tuples of length 2."
-            )
+        self.padding = _padding_init(padding, num_spatial_dims)
         self.dilation = dilation
         self.groups = groups
         self.use_bias = use_bias
+        self.padding_mode = _padding_mode_init(padding_mode)
+
+    def _nonzero_pad(self, x: Array) -> Array:
+        if isinstance(self.padding, str):
+            rhs_shape = tuple(
+                d * (k - 1) + 1 for k, d in zip(self.kernel_size, self.dilation)
+            )
+            padding = lax.padtype_to_pads(
+                x.shape[2:], rhs_shape, self.stride, self.padding
```

Comment:
Why `[2:]`?
`x` has shape `(in_channels, dim_1, ..., dim_N)`, so this is trimming off `dim_1` as well.

Reply:
A batch dimension is expanded in line 226, so the shape is `(batch, in_channels, dim_1 , ...)`.

Reply:
Ah, makes sense! Thanks.